docs/general/builtins/scalar-functions.rst

```diff
@@ -774,6 +774,64 @@ If you don't specify a time zone, ``truncate`` uses UTC time::
     +---------------+---------------+
     SELECT 3 rows in set (... sec)
 
+.. _date-bin:
+
+``date_bin(interval, timestamp, origin)``
+-----------------------------------------
+
+Return type is ``timestamp with time zone`` or ``timestamp without time zone``
+and same with timestamp and origin type.
+
+Puts timestamp to the beginning of interval aligned with origin. In case of
+single unit (1 second, 1 minute, 1 hour, 1 day) it gives the same result as
+:ref:`date_trunc <scalar-date-trunc>`. In case of 1 week interval date_bin
+returns same result as date_trunc only if origin is a Monday.
+
+If at least one argument is NULL, return value is NULL. Interval cannot be zero.
+Negative intervals are allowed and are treated the same as positive intervals.
+Intervals having month or year units are not supported due to varying length of
+those units.
+
+A timestamp can be binned to an interval of arbitrary length
+aligned with custom origin. 1609488000000 is Fri Jan 01 2021 08:00:00::
+
+    cr> select date_bin('3 hours' :: INTERVAL,
+    ... '2021-01-01T08:30:10Z'::timestamp without time zone,
+    ... '2021-01-01T05:00:00Z'::timestamp without time zone) as bin;
+    +---------------+
+    |           bin |
+    +---------------+
+    | 1609488000000 |
+    +---------------+
+    SELECT 1 row in set (... sec)
```

Comment:
Maybe we could use an example with `unnest` or multiple `VALUES` to better illustrate the effect of the binning?

Maybe something like this?


```
SELECT 
  date_bin('2 hours'::INTERVAL, ts, '2021-01-01T05:00:00Z'::TIMESTAMP) as bin, 
  date_format('%y-%m-%d %h:%i', date_bin('2 hours'::INTERVAL, ts, '2021-01-01T05:00:00Z'::TIMESTAMP)) formatted_bin
FROM unnest(ARRAY[
    '2021-01-01T08:30:10Z',
    '2021-01-01T08:38:10Z',
    '2021-01-01T18:18:10Z',
    '2021-01-01T18:18:10Z'
  ]::TIMESTAMP[]) as tbl (ts);
```